src/app/pages/directoryservice/idmap/idmap-form.component.ts

```diff
@@ -1,34 +1,35 @@
 import { Component } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
 import { MatDialog } from '@angular/material/dialog';
 import * as _ from 'lodash';
 import { FieldConfig } from '../../common/entity/entity-form/models/field-config.interface';
 import { FieldSet } from '../../common/entity/entity-form/models/fieldset.interface';
-import { ValidationService, IdmapService, DialogService } from '../../../services/';
+import { ValidationService, IdmapService, DialogService } from '../../../services';
 import { EntityJobComponent } from 'app/pages/common/entity/entity-job/entity-job.component';
-import { EntityUtils } from '../../../pages/common/entity/utils';
+import { EntityUtils } from '../../common/entity/utils';
 import helptext from '../../../helptext/directoryservice/idmap';
-
+import { ModalService } from '../../../services/modal.service';
+import { Subscription } from 'rxjs';
 @Component({
   selector: 'app-idmap-form',
   template: `<entity-form [conf]="this"></entity-form>`
 })
 export class IdmapFormComponent {
-
-  protected route_success: string[] = ['directoryservice', 'idmap'];
+  protected title: string;
   protected isEntity: boolean = true;
   protected namesInUse = [];
   protected queryCall = 'idmap.query';
   protected addCall = 'idmap.create';
   protected editCall = 'idmap.update';
-  protected queryCallOption: Array<any> = [["id", "="]];
+  protected pk: any;
+  protected queryKey = 'id';
+  private getRow = new Subscription;
```

Comment:
I wonder if this shouldn't just have the `Subscription` type (seen in other places too)?